packages/webview_flutter/webview_flutter_web/lib/src/web_webview_controller.dart

```diff
@@ -106,6 +110,37 @@ class WebWebViewController extends PlatformWebViewController {
       encoding: encoding,
     ).toString();
   }
+
+  @override
+  Future<void> addJavaScriptChannel(
+    JavaScriptChannelParams javaScriptChannelParams,
+  ) {
+    Null Function(html.Event) handler = (html.Event event) {
+      if (event is html.MessageEvent) {
+        javaScriptChannelParams.onMessageReceived(JavaScriptMessage(message: event.data));
+      }
+    };
+
+    javascriptChannels[javaScriptChannelParams.name] = handler;
+    html.window.addEventListener('message', handler);
+    
+    // TODO: what to return?
+    return Future.wait([]);
```

Comment:
@omartinma I'm not sure what to return here (and below).

Reply:
Hey @Frank3K, When using `Future<void>` there is no need to explicitly return anything.

Reply:
That's what I thought too, but when I remove the `return Future.wait([]);` I get the following error:

> The body might complete normally, causing 'null' to be returned, but the return type, `Future<void>`, is a potentially non-nullable type.
> Try adding either a return or a throw statement at the end. dart([body_might_complete_normally](https://dart.dev/diagnostics/body_might_complete_normally))

<img width="939" alt="image" src="https://user-images.githubusercontent.com/8014077/229486988-c1cdc2b4-5f09-4d86-9270-8ae1b7509218.png">

Reply:
@Frank3K  It could be because you are missing the `async` modifier in the function.

It should look like this: 
```
@override
  Future<void> addJavaScriptChannel(
    JavaScriptChannelParams javaScriptChannelParams,
  ) async {
 ...
}
```